### PR TITLE
Feature add weighted distribution

### PR DESCRIPTION
**Pull request by mel-flygare96**
Cards this pull request addresses:
Vision -- Create System for Applying Weighted Sampling to Light Input

To test:

Pull feature_add_weighted_distribution
Checkout master
Merge feature_add_weighted_distribution
Catkin make
Test on robot

If you review this PR, please add your name and email below.

Reviewers:
Reviewed-by: Nicholas Gilbert <gnick@pdx.edu>
Reviewed-by: Brady Testa <btesta@pdx.edu>

### CODE REVIEW

File: op3_ball_detector/src/ball_detector.cpp [hunk 1]
```diff
@@ -138,11 +142,19 @@ void BallDetector::process()
   if (enable_ == false)
     return;
 
-  if (switch_detection_flag_ == true && 50 <= num_call_++)
+  switch_detection_flag_ = true;
```
Comment (gilben1) [inline op3_ball_detector/src/ball_detector.cpp]: looks like this testing switch got left in (ln 145)

File: op3_ball_detector/src/ball_detector.cpp [hunk 2]
```diff
@@ -155,16 +167,17 @@ void BallDetector::process()
...
-    //std::cout << "TEST: " << img_filtered.empty() << std::endl;
-
     //detect circles
     houghDetection2(img_filtered);
   
     // was the detection valid?
     if (goodDetectionMode()) {
       std::cout << "Good detection mode for " << light_val << std::endl;
-      params_color_.adjustWeightsWithLightVal(light_val, 1, light_weights_);
+      params_color_.adjustWeightsWithLightVal(light_val, 4, light_weights_);
```
Comment (gilben1) [inline op3_ball_detector/src/ball_detector.cpp]: I'm fine with it, and it might depend on how you crafted the model, by why increase by four?

**Review by gilben1 — CHANGES_REQUESTED** [review]
See comment for line 145 in BallDetector.cpp

File: op3_ball_detector/src/ball_detector.cpp (hunk 1 above)
Comment (mel-flygare96) [reply op3_ball_detector/src/ball_detector.cpp]: Fixed. Also commented code for testing the distribution range. We can remove it once we have tested on the bot and determined it works.

**Review by gilben1 — APPROVED** [review]

**Comment by gilben1** [comment]
With the detection flag switched it looks good, approving pull request.

**Review by BradyT98 — APPROVED** [review]
The code looks correct to me, but I am curious as to the '4' parameter you passed as well. I suspect that number may just need to be fine tuned a bit here and there

File: op3_ball_detector/src/ball_detector.cpp (hunk 2 above)
Comment (mel-flygare96) [reply op3_ball_detector/src/ball_detector.cpp]: If the penalty and reward are the same, the distribution will never change unless it starts out really really good. Good color ranges are less common, and so are rewarded more. Bad ranges are more common, and are penalized less. Also, this means if the ball moves into a different light range, it won't take forever for the range to adjust to the new light level. However, these should be in constants. I will update them to be constants.

**Pull request merged**